### PR TITLE
added controller diagrams and a bit of theory on TECS

### PR DESCRIPTION
**Pull request by RomanBapst**
Together with @sfuhrer we've been working on adding controller diagrams and some theoretical documentation for TECS.

@hamishwillee Initially I was going to add the tuning guide for TECS right after this but as discussed with @sfuhrer it makes more sense to have that in the user guide.

Signed-off-by: Roman <bapstroman@gmail.com>

### CODE REVIEW

File: LANGS.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
 * [English](en/)
-* [Chinese (中文)](zh/)
+<!-- * [Chinese (中文)](zh/)
```
Comment (hamishwillee) [inline LANGS.md]: Undo. This change turns off our translation for Chinese and Korean on the live site. I sometimes do this to reduce build times locally, but you shouldn't save it to master.

File: book.json [hunk 2]
```diff
@@ -19,8 +19,8 @@
         "validate-links",
         "versions-select@git+https://github.com/Dronecode/gitbook-plugin-versions-select.git",
         "mathjax@git+https://github.com/Dronecode/plugin-mathjax.git#update_cdn",
-        "mermaid",
-        "-stub-out-blocks@git+https://github.com/hamishwillee/gitbook-plugin-stub-out-blocks.git",
+        "-mermaid",
```
Comment (hamishwillee) [inline book.json]: Undo this change. Needed for local testing, but don't push live.

File: assets/diagrams/TECS_throttle.html [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
```
Comment (hamishwillee) [inline assets/diagrams/TECS_throttle.html]: What is this file for? It isn't imported or linked, and generally HTML files should not be used directly, and certainly not as assets in diagrams dir.

File: en/flight_stack/controller_diagrams.md [hunk 4]
```diff
@@ -16,6 +16,62 @@ Request access from dev team. -->
 * Depending on the mode, the outer (position) loop is bypassed (shown as a multiplexer after the outer loop). The position loop is only used when holding position or when the requested velocity in an axis is null.
 * The integrator in the inner loop (velocity) controller includes an anti-reset windup (ARW) using a clamping method.
 
+## Fixed-Wing Position Controller
+
+### Total Energy Control System (TECS)
+The PX4 implementation of the Total Energy Control System (TECS) enables simultaneous control of true airspeed and altitude of a fixed wing aircraft.
+The code is implemented as a library which is used in the fixed wing position control module (make a reference here). As seen in the diagram below, TECS receives as inputs airspeed and altitude setpoints and outputs a throttle and pitch angle setpoint. These two outputs are sent to the fixed wing attitude controller which implements the attitude control solution. It's therfore important to understand that the performance of TECS is directly affected by the performance of the pitch control loop. A poor tracking of airspeed and altitude is often caused by a poor tracking of the aircraft pitch angle.
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md]: Missing reference "(make a reference here)" - where should this go?

Comment (hamishwillee) [reply en/flight_stack/controller_diagrams.md]: I'd be tempted to move the diagram to just before your text "As seen in the diagram below," (and of course changing that from "below" to "above". This effectively shows your point, which is that TECS is part of the Fixed Wing Position control module.

File: en/flight_stack/controller_diagrams.md [hunk 5]
```diff
@@ -16,6 +16,62 @@ Request access from dev team. -->
 * Depending on the mode, the outer (position) loop is bypassed (shown as a multiplexer after the outer loop). The position loop is only used when holding position or when the requested velocity in an axis is null.
 * The integrator in the inner loop (velocity) controller includes an anti-reset windup (ARW) using a clamping method.
 
+## Fixed-Wing Position Controller
+
+### Total Energy Control System (TECS)
+The PX4 implementation of the Total Energy Control System (TECS) enables simultaneous control of true airspeed and altitude of a fixed wing aircraft.
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md:22]: Typo "therfore"

File: en/flight_stack/controller_diagrams.md [hunk 6]
```diff
@@ -16,6 +16,62 @@ Request access from dev team. -->
...
+The code is implemented as a library which is used in the fixed wing position control module (make a reference here). As seen in the diagram below, TECS receives as inputs airspeed and altitude setpoints and outputs a throttle and pitch angle setpoint. These two outputs are sent to the fixed wing attitude controller which implements the attitude control solution. It's therfore important to understand that the performance of TECS is directly affected by the performance of the pitch control loop. A poor tracking of airspeed and altitude is often caused by a poor tracking of the aircraft pitch angle.
+
+> **Note** Make sure to tune the attitude controller before attempting to tune TECS.
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md]: Good comment. Is there a link to tuning the attitude controller? - or is this the doc you still plan to write?

File: en/flight_stack/controller_diagrams.md (hunk 4 above)
Comment (hamishwillee) [reply en/flight_stack/controller_diagrams.md]: Typo "tranforming "

File: en/flight_stack/controller_diagrams.md [hunk 7]
```diff
@@ -16,6 +16,62 @@ Request access from dev team. -->
 * Depending on the mode, the outer (position) loop is bypassed (shown as a multiplexer after the outer loop). The position loop is only used when holding position or when the requested velocity in an axis is null.
 * The integrator in the inner loop (velocity) controller includes an anti-reset windup (ARW) using a clamping method.
 
+## Fixed-Wing Position Controller
```
Comment (hamishwillee) [inline en/flight_stack/controller_diagrams.md:19]: This looks excellent - I've only scanned the maths, but the explanation of TECS and the control problem is very clear.

TECS is part of the Fixed Wing Position Control module. IFF it is the only part that you intend to document here then the current structure is "OK". If you ever expect to cover the other bits then I would restructure slightly to have the diagram and a description of Fixed Wing Position Control under the parent heading. This makes it easier to provide context for TECs and any other components. So something like:

```
## Fixed-Wing Position Controller

 The fixed wing position control module is used to set vehicle attitude based on desired X, Y, Z. It uses the TECs library to XXX, and the L1 controller to ZZZ. The structure is shown below.

	![TECS](../../assets/diagrams/tecs_in_context.svg)

### Total Energy Control System (TECS)

...
```
Hope that makes sense.

**Comment by hamishwillee** [comment]
@RomanBapst @sfuhrer Great job. Comments added.

> I was going to add the tuning guide for TECS right after this but as discussed with @sfuhrer it makes more sense to have that in the user guide.

I agree - this is our current docs model. 

In future it might make sense to have these sorts of diagrams/discussion in the tuning guide(s). Depends on whether you need them as an "end user" tuning your vehicle, or they are only useful if you are a developer and want to create a new type of frame.

File: en/flight_stack/controller_diagrams.md (hunk 6 above)
Comment (RomanBapst) [reply en/flight_stack/controller_diagrams.md]: @hamishwillee Not sure if there is. If yes, I will add it. Otherwise it's something for the future.

Comment (hamishwillee) [reply en/flight_stack/controller_diagrams.md]: OK then. I'm going to merge this now!
